src/code/models/codap-connect.ts

```diff
@@ -780,6 +799,61 @@ export class CodapConnect {
     });
     return promise;
   }
+
+  public showGuideItemAtIndex(index: number) {
+    if (this.guideComponentId) {
+      this.codapPhone.call({
+        action: "update",
+        resource: `component[${this.guideComponentId}]`,
+        values: {
+          currentItemIndex: index,
+          isVisible: true
+        }
+      }, (result) => {
+        console.log("showGuideItemAtIndex", result);
+      });
+    }
+  }
+
+  private getGuideItems() {
+    const getGuideComponent = (id) => {
+      this.codapPhone.call({
+        action: "get",
+        resource: `component[${id}]`
+      }, (result) => {
+        if (result && result.success && result.values) {
+          const items = result.values.items || [];
+          AppSettingsActions.setGuideItems(items);
+        }
+      });
+    };
+
+    if (!this.guideComponentId) {
+      this.codapPhone.call({
+        action: "get",
+        resource: "componentList"
+      }, (result) => {
+        if (result && result.success && result.values) {
+          _.map(result.values, (value) => {
+            if (value.type === "guideView") {
+              this.guideComponentId = value.id;
+            }
+          });
+          if (this.guideComponentId) {
+            getGuideComponent(this.guideComponentId);
+          }
+        }
+      });
+    } else {
+      getGuideComponent(this.guideComponentId);
+    }
+  }
+
+  private pollForGuideItemUpdates() {
+    if (!this.guidePollerInterval) {
+      this.guidePollerInterval = window.setInterval(() => this.getGuideItems(), 1000);
```

Comment:
Not knowing how this API is meant to be used, I have a question:  Could the guideItems change?  Do we have to poll forever, or should we eventually be clearing the interval?

Reply:
We have to poll forever once the guide configuration gui is opened - CODAP doesn't give us any callbacks to find out when the gui is closed.  I setup the poller this way because it will be a rare occurance to configure the guides so the poller is only enabled when the gui is opened.

Reply:
Did you discuss this with @jsandoe ? A notification on close seems like a reasonable request for addition to the plugin API.